pages/dashboard/workspaces/[workspaceid]/collections/[collectionid]/settings/creators.vue

```diff
@@ -66,65 +76,131 @@ if (error.value) {
 }
 
 if (data.value) {
-  moduleData.creators = normalizeCreators(data.value);
+  creators.value = normalizeCreators(data.value);
 }
 
-const addCreator = () => {
-  moduleData.creators.push({
+const openAddCreatorDrawer = () => {
+  selectedCreator.value = {
     affiliation: "",
-    creatorIndex: moduleData.creators.length + 1,
+    creatorIndex: creators.value.length + 1,
     creatorName: "",
     familyName: "",
     givenName: "",
     identifier: "",
     identifierType: null,
     nameType: "Personal",
+  };
+
+  drawerAction.value = "Add";
+  showCreatorDrawer.value = true;
+};
+
+const openEditCreatorDrawer = (index: number) => {
+  const creator = creators.value[index];
+
+  if (creator) {
+    selectedCreator.value = {
+      affiliation: creator.affiliation,
+      creatorIndex: creator.creatorIndex,
+      creatorName: creator.creatorName,
+      familyName: creator.familyName,
+      givenName: creator.givenName,
+      identifier: creator.identifier,
+      identifierType: creator.identifierType,
+      nameType: creator.nameType,
+    };
+
+    drawerAction.value = "Edit";
+    showCreatorDrawer.value = true;
+  } else {
+    push.error("Something went wrong");
+  }
+};
+
+const deleteCreator = (index: number) => {
+  const newCreators = creators.value.filter(
+    (creator) => creator.creatorIndex !== index,
+  );
+
+  creators.value = normalizeCreators(newCreators);
+
+  saveCreators();
+};
+
+const saveCreators = async () => {
+  const data = creators.value.map((creator, index) => {
+    return {
+      ...creator,
+      creatorIndex: index,
+      identifierType: creator.identifierType || "",
+    };
   });
+
+  loading.value = true;
+
+  await $fetch(
+    `/api/workspaces/${workspaceid}/collections/${collectionid}/creators`,
+    {
+      body: JSON.stringify(data),
+      headers: useRequestHeaders(["cookie"]),
+      method: "PUT",
+    },
+  )
+    .then((_response) => {
+      console.log("success");
+
+      push.success("Creators updated successfully.");
+
+      creators.value = normalizeCreators(data); // todo: might not be needed
+      showCreatorDrawer.value = false;
+    })
+    .catch((error) => {
+      console.error(error);
+      push.error("Something went wrong.");
+    })
+    .finally(() => {
+      loading.value = false;
+    });
 };
 
-const saveCreators = (e: MouseEvent) => {
+const confirmEdits = (e: MouseEvent) => {
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_clarification):** Potential confusion between `saveCreators` and `confirmEdits`.

The `confirmEdits` function seems to be doing more than confirming edits; it also updates or adds creators. Consider renaming it to better reflect its functionality or splitting it into more focused functions.

```suggestion
const updateOrCreateCreators = (e: MouseEvent) => {
```